lib/jxl/enc_ans.cc

```diff
@@ -1556,26 +1613,55 @@ size_t BuildAndEncodeHistograms(const HistogramParams& params,
     }
   }
 
-  bool use_prefix_code =
-      params.force_huffman || total_tokens < 100 ||
-      params.clustering == HistogramParams::ClusteringType::kFastest ||
-      ans_fuzzer_friendly_;
-  if (!use_prefix_code) {
-    bool all_singleton = true;
-    for (size_t i = 0; i < num_contexts; i++) {
-      if (builder.Histo(i).ShannonEntropy() >= 1e-5) {
-        all_singleton = false;
+  if (params.add_missing_symbols) {
+    for (size_t c = 0; c < num_contexts; ++c) {
+      for (int symbol = 0; symbol < ANS_MAX_ALPHABET_SIZE; ++symbol) {
+        builder.VisitSymbol(symbol, c);
       }
     }
-    if (all_singleton) {
-      use_prefix_code = true;
+  }
+
+  if (params.update_global_state) {
+    bool use_prefix_code =
+        params.force_huffman || total_tokens < 100 ||
+        params.clustering == HistogramParams::ClusteringType::kFastest ||
+        ans_fuzzer_friendly_;
+    if (!use_prefix_code) {
+      bool all_singleton = true;
+      for (size_t i = 0; i < num_contexts; i++) {
+        if (builder.Histo(i).ShannonEntropy() >= 1e-5) {
+          all_singleton = false;
+        }
+      }
+      if (all_singleton) {
+        use_prefix_code = true;
+      }
     }
+    codes->use_prefix_code = use_prefix_code;
+  }
+
+  if (params.add_fix_histograms) {
+    // TODO(szabadka) Add more fixed histograms.
+    const size_t alphabet_size = ANS_MAX_ALPHABET_SIZE;
```

Comment:
Presumably, we can get away with smaller alphabets if we decide we are ok with, say, a 000 uint config (then 32 symbols ought to be more than enough)

Reply:
I added a separate TODO for this